### PR TITLE
Enhance pp tabset

### PR DESCRIPTION
**Pull request by triitvn**

### CODE REVIEW

File: src/directives/pp-tabset/pp-tab-heading.directive.js [hunk 1]
```diff
@@ -10,27 +10,26 @@
...
         bindToController: true,
-        controller: ['PP_TABSET_CONSTANTS', ppTabHeadingController],
+        controller: [ppTabHeadingController],
         require: ['^^ppTabset', '^ppTab', 'ppTabHeading'],
         link: function(scope, element, attrs, ctrls) {
           var ppTabsetCtrl = ctrls[0];
           var ppTabHeadingCtrl = ctrls[2];
 
-          ppTabHeadingCtrl.styleName = ppTabsetCtrl.getStyleName();
+          ppTabHeadingCtrl.headingTemplateUrl =
```
Comment (nlnbich) [inline src/directives/pp-tabset/pp-tab-heading.directive.js]: can access [headingTemplateUrl] from tabConfig ?

File: less/utils/mixins.less [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.mixinArrowDown(@width, @height, @color) {
+  width: 0px;
```
Comment (nlnbich) [inline less/utils/mixins.less]: remove [px]
for css, use [0] dont need unit

File: less/utils/mixins.less [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.mixinArrowDown(@width, @height, @color) {
```
Comment (nlnbich) [inline less/utils/mixins.less]: for arrow, do we really need width & height ??
-create util class for: up, down, left, right (at least up & down)

File: src/directives/pp-tabset/pp-tab-heading.directive.js [hunk 4]
```diff
@@ -10,27 +10,26 @@
     .directive('ppTabHeading', function() {
       return {
         restrict: 'E',
```
Comment (nlnbich) [inline src/directives/pp-tabset/pp-tab-heading.directive.js]: should use [replace: true]

File: src/directives/pp-tabset/pp-tab-heading.html [hunk 5]
```diff
@@ -1,20 +1,18 @@
...
-        disabled="!vm.selected"
-      ></pp-icon>
-    </div>
-    
-    <div class="pp-tab-heading-text" ng-bind-html="vm.tabConfig.headingText"></div>
-
-    <div ng-show="vm.tabConfig.headingHelpMessage" class="pp-tab-heading-right-icon">
-      <pp-help-icon>{{vm.tabConfig.headingHelpMessage}}</pp-help-icon>
-    </div>
+<div ng-if="vm.headingTemplateUrl === false" class="pp-tab-heading-content-wrapper">
```
Comment (nlnbich) [inline src/directives/pp-tabset/pp-tab-heading.html]: !vm.headingTemplateUrl

File: src/directives/pp-tabset/pp-tab-heading.html [hunk 6]
```diff
@@ -1,20 +1,18 @@
...
-    <div class="pp-tab-heading-text">{{vm.tabConfig.headingText}}</div>
+  <div ng-if="vm.tabConfig.headingHelpMessage" class="pp-tab-heading-right-icon">
+    <pp-help-icon>{{vm.tabConfig.headingHelpMessage}}</pp-help-icon>
   </div>
-</div>
\ No newline at end of file
+</div>
+
+<div 
+  ng-if="vm.headingTemplateUrl !== false"
```
Comment (nlnbich) [inline src/directives/pp-tabset/pp-tab-heading.html]: vm.headingTemplateUrl

File: src/directives/pp-tabset/pp-tab.html [hunk 7]
```diff
@@ -9,7 +9,8 @@
   <uib-tab-heading class="pp-tab-heading-content-wrapper">
     <pp-tab-heading 
       tab-config="vm.tabConfig"
-      selected="vm.selected"></pp-tab-heading>
+      selected="vm.selected"
+      parent-scope="vm.tabsetConfig.parentScope"></pp-tab-heading>
```
Comment (nlnbich) [inline src/directives/pp-tabset/pp-tab.html]: send [tabSetConfig] intead

File: src/directives/pp-tabset/pp-tabset.constants.js [hunk 8]
```diff
@@ -1,40 +1,30 @@
...
 
-  var STYLE_NAMES = {
-    DEFAULT: 'DEFAULT',
-    SIMPLE_UNDERLINED: 'SIMPLE_UNDERLINED',
-    NONE: 'NONE'
-  };
-
   var EVENTS = {
-    DID_SELECT_TAB: 'pp-tabset-did-select-tab'
+    didSelectedTab: 'pp-tabset-did-select-tab'
```
Comment (nlnbich) [inline src/directives/pp-tabset/pp-tabset.constants.js]: use meaningful name

File: src/directives/pp-tabset/pp-tabset.directive.js [hunk 9]
```diff
@@ -127,7 +128,7 @@
       var deselectedPpTab = getPpTab(deselectedTabIndex);
       _.set(deselectedPpTab, ['controller', 'selected'], false);
```
Comment (nlnbich) [inline src/directives/pp-tabset/pp-tabset.directive.js]: why tabController need to save [selected] stated ?

File: src/directives/pp-tabset/pp-tab-heading.html [hunk 10]
```diff
@@ -1,20 +1,18 @@
...
-    <div class="pp-tab-heading-text" ng-bind-html="vm.tabConfig.headingText"></div>
-
-    <div ng-show="vm.tabConfig.headingHelpMessage" class="pp-tab-heading-right-icon">
-      <pp-help-icon>{{vm.tabConfig.headingHelpMessage}}</pp-help-icon>
-    </div>
+<div ng-if="vm.headingTemplateUrl === false" class="pp-tab-heading-content-wrapper">
+  <div ng-if="vm.tabConfig.headingLeftIconName" class="pp-tab-heading-left-icon">
+    <pp-icon 
+      name="{{vm.tabConfig.headingLeftIconName}}"
+      disabled="!vm.selected"
```
Comment (nlnbich) [inline src/directives/pp-tabset/pp-tab-heading.html]: why just first [pp-icon] has [disabled] attribute ?

File: src/directives/pp-tabset/pp-tabset.directive.js [hunk 11]
```diff
@@ -67,11 +68,11 @@
     function getPpTabsetClassNames(tabsetConfig) {
       var tabsetConfigSize = _.get(tabsetConfig, 'size');
       var tabsetStyle = getTabsetStyle();
-      var tabsetSize = _.get(tabsetStyle, ['SIZES', tabsetConfigSize]);
+      var tabsetSize = _.get(tabsetStyle, ['sizes', tabsetConfigSize]);
 
       return _.concat(
```
Comment (nlnbich) [inline src/directives/pp-tabset/pp-tabset.directive.js]: - do we need `|| []` ??
- _.concat always has value, at least [undefined, undefined]

File: src/directives/pp-tabset/pp-tab-heading.html (hunk 5 above)
Comment (triitvn) [inline src/directives/pp-tabset/pp-tab-heading.html]: Updated

File: less/utils/mixins.less (hunk 2 above)
Comment (triitvn) [inline less/utils/mixins.less]: Updated

File: src/directives/pp-tabset/pp-tab-heading.html (hunk 6 above)
Comment (triitvn) [inline src/directives/pp-tabset/pp-tab-heading.html]: Updated

File: src/directives/pp-tabset/pp-tab.html (hunk 7 above)
Comment (triitvn) [inline src/directives/pp-tabset/pp-tab.html]: Updated

File: src/directives/pp-tabset/pp-tabset.constants.js (hunk 8 above)
Comment (triitvn) [inline src/directives/pp-tabset/pp-tabset.constants.js]: updated: `onTabChanged`

File: src/directives/pp-tabset/pp-tabset.directive.js (hunk 11 above)
Comment (triitvn) [inline src/directives/pp-tabset/pp-tabset.directive.js]: Updated: 
```
return _.concat(
        _.get(tabsetStyle, 'className', []),
        _.get(tabsetSize, 'className', [])
);
```

File: src/directives/pp-tabset/pp-tabset.directive.js (hunk 9 above)
Comment (triitvn) [inline src/directives/pp-tabset/pp-tabset.directive.js]: Removed selected flag.
Added:
```
ppTabCtrl.isSelected = isSelected;
function isSelected() {
    return _.get(ppTabsetCtrl, 'selectedTabIndex') === ppTabCtrl.index;
}
```

File: src/directives/pp-tabset/pp-tab-heading.directive.js (hunk 4 above)
Comment (triitvn) [inline src/directives/pp-tabset/pp-tab-heading.directive.js]: Will be deprecated in Component https://docs.angularjs.org/guide/component

File: src/directives/pp-tabset/pp-tab-heading.directive.js (hunk 1 above)
Comment (triitvn) [inline src/directives/pp-tabset/pp-tab-heading.directive.js]: Updated

File: less/utils/mixins.less (hunk 3 above)
Comment (triitvn) [inline less/utils/mixins.less]: Moved to pp-tabset-2

**Pull request merged**